### DISCUSSION TITLE
How to configure multiple environment builds?

### DISCUSSION THREAD

**Discussion by peidongGuo** [web-infra-dev/rsbuild#1199] [Q&A] [answered]
怎么配置多套环境编译？
比如，我有 local、development、test、production 等还有些私有化部署的各种环境配置文件，我怎么在命令行里加载不同环境配置文件？

**Comment by chenjiahan** [comment, marked as answer]
可以参考以下文档：

- https://rsbuild.dev/zh/guide/basic/config#%E6%8C%87%E5%AE%9A%E9%85%8D%E7%BD%AE%E6%96%87%E4%BB%B6
- https://rsbuild.dev/zh/guide/basic/config#%E4%BD%BF%E7%94%A8%E7%8E%AF%E5%A2%83%E5%8F%98%E9%87%8F

**Reply by Zuojiangtao** [reply]
404了

**Reply by Zuojiangtao** [reply]
https://rsbuild.dev/zh/guide/advanced/env-vars#env-%E6%A8%A1%E5%BC%8F

**Comment by peidongGuo** [comment]
另外，还有个问题：
packages 中 scripts 我这样设置，
`"build:development": "NODE_ENV=development rsbuild build"`
rsbuild.config.ts 如下设置，
```
output: {
    assetPrefix: '/test',
    cssModules: { localIdentName: '' },
  },

```
index.html 模板这样设置
   ` <link rel="icon" href="<%= assetPrefix %>/favicon.ico" />`

最后编译后：
 <link rel="icon" href="/favicon.ico" />

推断一个原因： NODE_ENV 设置后，output.asstPrefix 不起作用了。

**Reply by chenjiahan** [reply]
不太推荐在执行 build 命令的时候设置 `NODE_ENV=development`，如果设置了这个值，Rsbuild 会走到 dev 环境的逻辑，也就是使用 [dev.assetPrefix](https://rsbuild.dev/config/dev/asset-prefix) 选项。

**Reply by peidongGuo** [reply]
好的。那现在确实是要使用多个配置文件来解决这个问题。

**Reply by Ru1n33** [reply]
> 不太推荐在执行 build 命令的时候设置 `NODE_ENV=development`，如果设置了这个值，Rsbuild 会走到 dev 环境的逻辑，也就是使用 [dev.assetPrefix](https://rsbuild.dev/config/dev/asset-prefix) 选项。

你好，如果我想部署在测试环境下，npm run build:stage这种，如何让assetPrefix生效呢，不生效的话，静态资源都无法加载出来，有什么解决方案
